MediaBrowser.Api/LiveTv/LiveTvService.cs

```diff
@@ -866,10 +868,30 @@ private void AssertUserCanManageLiveTv()
 
         public async Task<object> Post(AddListingProvider request)
         {
+            if (request.Pw != null)
+            {
+                request.Password = GetHashedString(request.Pw);
+            }
+
+            request.Pw = null;
+
             var result = await _liveTvManager.SaveListingProvider(request, request.ValidateLogin, request.ValidateListings).ConfigureAwait(false);
             return ToOptimizedResult(result);
         }
 
+        /// <summary>
+        /// Gets the hashed string.
+        /// </summary>
+        private string GetHashedString(string str)
+        {
+            // SchedulesDirect requires a SHA1 hash of the user's password
+            // https://github.com/SchedulesDirect/JSON-Service/wiki/API-20141201#obtain-a-token
+            SHA1 sha = SHA1.Create();
+            return BitConverter.ToString(
```

Comment:
We have a ‘HexHelper’ class for this, you should use that.

Reply:
Fixed